### PR TITLE
#16 - Update text component 

### LINKED ISSUE #16: Update text component
- Support Internationalization

Props:
- tx: string - i18n Key
- txOptions?: Object 
- color: Color
- text: string - Plain text

These properties must always override the variants

### PR DESCRIPTION
**Pull request by Koushith**
### Description

- [x] Removed TextPreset and made it as a Variant.
- [x] Tested all Alignment Props and it is working as expected.
- [x] Tested all Custom Fonts and styling props.
- [x] Moved the switch case out of the main Text component.
- [x] Refactored code by removing unused variables and destructured props.

### CODE REVIEW

File: src/components/primitive/text/text.component.props.ts [hunk 1]
```diff
@@ -1,67 +1,55 @@
...
-  MESSAGE = 'message',
+  CONTENT = 'content',
 
   /**
    * Renders a small gray text
    */
   SMALL = 'small',
 }
 
+export enum Variant {
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.props.ts]: Same data cannot originate at 2 places. Use TextPresets itself with lowercase property names if required.

File: src/components/primitive/text/text.component.props.ts [hunk 2]
```diff
@@ -158,26 +135,15 @@ export interface TextProps extends React.HTMLAttributes<HTMLHeadingElement> {
...
-   * While appending or prepending a text, a spacing is automatically added.
-   * If noSpace prop is true, then the space will not be added.
-   */
-  noSpace?: boolean;
-
   /**
    * One of the different types of text presets.
    */
-  preset?: TextPresets;
+  preset?: string;
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.props.ts:147]: What's the difference between preset and a variant? Shouldn't these be same?

File: src/components/primitive/text/text.component.props.ts [hunk 3]
```diff
@@ -1,67 +1,55 @@
...
  */
-export const FontSizes = {
-  tiny: 1.2,
-  small: 1.6,
-  medium: 2.0,
-  large: 2.6,
-};
-
-type Size = keyof typeof FontSizes;
+export enum FontSizes {
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.props.ts]: Don't use plural names for enums. It should be `FontSize`.

File: src/components/primitive/text/text.component.props.ts [hunk 4]
```diff
@@ -158,26 +135,15 @@ export interface TextProps extends React.HTMLAttributes<HTMLHeadingElement> {
...
    * One of the different types of text presets.
    */
-  preset?: TextPresets;
+  preset?: string;
 
   /**
    * Font size from the predefined set of sizes or an arbitrary number
    */
-  size?: Size | number;
+  size?: keyof typeof FontSizes | number;
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.props.ts]: FontSizes is already a type. Have you tried `keyof FontSizes`?

File: src/components/primitive/text/text.component.props.ts [hunk 5]
```diff
@@ -188,7 +154,16 @@ export interface TextProps extends React.HTMLAttributes<HTMLHeadingElement> {
...
+   * determines the differnt text variants.
+   */
+
+  variant?: keyof typeof Variant;
+
+  /**
+   * sets the font weight
+   */
+
+  weight?: keyof typeof FontWeight;
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.props.ts:174]: typeof Required?

File: src/components/primitive/text/text.component.tsx [hunk 6]
```diff
@@ -1,77 +1,77 @@
...
+      case TextPresets.TITLE:
+        TextComponent = Title;
+        break;
+      case TextPresets.CONTENT_HEADER:
+        TextComponent = ContentHeader;
+        break;
+      case TextPresets.CONTENT:
+        TextComponent = Content;
+        break;
+
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.tsx]: Line break required?

File: src/components/primitive/text/text.component.tsx [hunk 7]
```diff
@@ -1,77 +1,77 @@
...
-      {children}
-    </Typography>
-  );
-};
+      text-transform: ${({ upperCase }) => (upperCase ? `uppercase` : '')};
+      text-transform: ${({ lowerCase }) => (lowerCase ? `lowercase` : '')};
+      text-transform: ${({ capitalize }) => (capitalize ? `lowercase` : '')};
+      text-align: ${({ center }) => (center ? `center` : '')};
+      text-align: ${({ left }) => (left ? `left` : '')};
+      text-align: ${({ right }) => (right ? `right` : '')};
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.tsx]: Tested alignments properly?

File: src/components/primitive/text/text.component.tsx [hunk 8]
```diff
@@ -1,77 +1,77 @@
...
+      line-height: ${({ lineHeight }) => (lineHeight ? `${lineHeight}em` : '')};
+      padding-left: ${({ paddingLeft }) => (paddingLeft ? ` ${paddingLeft}rem` : '')};
+      padding-right: ${({ paddingRight }) => (paddingRight ? ` ${paddingRight}rem` : '')};
+      font-weight: ${({ noBold }) => (noBold ? `normal` : '')};
+      font-weight: ${({ bold600 }) => (bold600 ? `600` : '')};
+      font-size: ${({ size }) => (size ? `${FontSizes[size]}rem` : '')};
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.tsx]: Have you tested custom font size?

File: src/i18n/locales/en.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+  "signIn": "Sign In ",
+  "signOut": "Sign Out",
+  "Home": "Home",
+  "Claims": "Claims",
```
Comment (dsouzaedison) [inline src/i18n/locales/en.json]: Property names should be camel-cased. Also, keys need to be grouped. The keys mentioned here do not seem to be relevant to this ticket.

File: src/components/primitive/text/text.component.tsx [hunk 10]
```diff
@@ -1,77 +1,77 @@
...
+      color: ${colors[color]} !important;
+    `;
 
-export { Text };
+    return <Typography {...rest}>{content}</Typography>;
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.tsx:55]: Rest props are passed only when the styled component doesn't have access to component props. Please move the Typography component out if you are using this syntax.

**Review by dsouzaedison — CHANGES_REQUESTED** [review]

File: src/components/primitive/text/text.component.styles.ts [hunk 11]
```diff
@@ -1,32 +1,90 @@
...
+    text-transform: ${({ capitalize }) => capitalize && `lowercase`};
+    text-align: ${({ center }) => center && 'center'};
+    text-align: ${({ left }) => left && `left`};
+    text-align: ${({ right }) => right && `right`};
+    line-height: ${({ lineHeight }) => lineHeight && `${lineHeight}em`};
+    padding-left: ${({ paddingLeft }) => paddingLeft && ` ${paddingLeft}rem`};
+    padding-right: ${({ paddingRight }) => paddingRight && ` ${paddingRight}rem`};
+    font-weight: ${({ noBold }) => noBold && `normal`};
+    font-weight: ${({ bold500 }) => bold500 && `500`};
+    font-weight: ${({ bold600 }) => bold600 && `600`};
```
Comment (dsouzaedison) [inline src/components/primitive/text/text.component.styles.ts]: How style will be rendered if bold600 is false?

Comment (Koushith) [reply src/components/primitive/text/text.component.styles.ts]: Those are optional props used for custom styling. if it is false, the default component will be rendered.

**Review by dsouzaedison — APPROVED** [review]
